app/com/linkedin/drelephant/exceptions/core/ExceptionFingerprintingRunner.java

```diff
@@ -99,20 +115,103 @@ private void saveDriverExceptionLogForExceptionFingerPrinting(List<ExceptionInfo
+      //sparkJobException.exceptionLog = exceptionsTrace;
+      if (exceptionsTrace.trim().length() > 2) {
+        sparkJobException.exceptionLog = exceptionsTrace;
       } else {
         sparkJobException.exceptionLog = "Couldn't gather driver logs for the job";
       }
       sparkJobException.exceptionType = ExceptionInfo.ExceptionSource.DRIVER.toString();
-      sparkJobException.save();
+      saveSparkJobException(sparkJobException, exceptionInfoList);
       jobsExceptionFingerPrinting.save();
     }
   }
 
+  private void saveSparkJobException(JobsExceptionFingerPrinting sparkJobException,
+      List<ExceptionInfo> exceptionInfoList) {
+    try {
+      sparkJobException.save();
+      logger.debug(" Final String to be saved in dB " + sparkJobException.exceptionLog);
+    } catch (PersistenceException pe) {
+      logger.error(" Error while storing spark job exception ", pe);
+      List<ExceptionInfo> halfExceptionList = createMixofException(exceptionInfoList, exceptionInfoList.size() / 2);
+      sparkJobException.exceptionLog = parseExceptions(halfExceptionList, false);
+      try {
+        sparkJobException.save();
+        logger.debug(" Final String to be saved in dB " + sparkJobException.exceptionLog);
+      } catch (PersistenceException pe1) {
+        logger.error(" Error while storing 1/2 of the spark job exception ", pe1);
+        List<ExceptionInfo> firstException = createMixofException(exceptionInfoList, 2);
+        sparkJobException.exceptionLog = parseExceptions(firstException, false);
+        sparkJobException.save();
+        logger.debug(" Final String to be saved in dB " + sparkJobException.exceptionLog);
+      }
+    }
+  }
+
+  /**
+   * Since we are not able to store all the exceptions in database , this method
+   * will create the mix of exceptions (including Executor and Driver)
+   * @param exceptionInfoList : List of exceptions .
+   * @param size : Maximum size of the list
+   * @return : Mix of exceptions
+   */
+  private List<ExceptionInfo> createMixofException(List<ExceptionInfo> exceptionInfoList, int size) {
+    List<ExceptionInfo> mixOfException = new ArrayList<>();
+    List<ExceptionInfo> executorExceptions = new ArrayList<>();
+    for (ExceptionInfo exceptionInfo : exceptionInfoList) {
+      if (exceptionInfo.getExceptionSource().name().equals(ExceptionInfo.ExceptionSource.EXECUTOR.name())) {
+        executorExceptions.add(exceptionInfo);
+      }
+    }
+    logger.info(" Size of executor exceptions "+executorExceptions.size());
+    for (int index = 0; index <= size / 2; index++) {
+      mixOfException.add(exceptionInfoList.get(index));
+    }
+    logger.info(" Size of mix exceptions "+mixOfException.size());
+    int max = Math.min(executorExceptions.size(), size - (size / 2));
+    for (int index = 0; index < max; index++) {
+      mixOfException.add(executorExceptions.get(index));
+    }
+    logger.info(" Size of mix exceptions "+mixOfException.size());
```

Comment:
Seems like there is a chance of duplication of Executor type exceptions in mixOfException. But if there is a de-duplication done or it is guaranteed that `exceptionInfoList` is ordered in a way that Executor type of exceptions will be found in the last half of the `exceptionInfoList` then will be fine.

Reply:
Yes exceptionInfoList  its being made sure that executor exceptions will be last half of exceptionInfoList .  It's being made in parseException method that collections are order by weight . All the driver exception have higher weight then executor excpetions .